### PR TITLE
Fix windows storage tests

### LINKED ISSUE #103651: [Failing Test] [sig-storage] subPath should unmount if pod is deleted while kubelet is down (ci-kubernetes-e2e-gci-gce-serial)
#### Which jobs are failing:
ci-kubernetes-e2e-gci-gce-serial (gce-cos-master-serial)

#### Which test(s) are failing:
Four `Kubernetes e2e suite.[sig-storage] * Volumes [Driver: *] [Testpattern: *] subPath should unmount if pod is * deleted while kubelet is down [Disruptive][Slow][LinuxOnly]` tests:

* Kubernetes e2e suite.[sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (default fs)] subPath should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow][LinuxOnly]
* Kubernetes e2e suite.[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow][LinuxOnly]
* Kubernetes e2e suite.[sig-storage] In-tree Volumes [Driver: hostPathSymlink] [Testpattern: Inline-volume (default fs)] subPath should unmount if pod is gracefully deleted while kubelet is down [Disruptive][Slow][LinuxOnly]
* Kubernetes e2e suite.[sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (default fs)] subPath should unmount if pod is force deleted while kubelet is down [Disruptive][Slow][LinuxOnly]

#### Since when has it been failing:
`Kubernetes e2e suite.[sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (default fs)] subPath should unmount if pod is force deleted while kubelet is down [Disruptive][Slow][LinuxOnly]` is failing since June 29th 2021 (but the first failure was different than more recent ones). Commit 1151dc1ee, infra commit a97a41ecc, [build](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/1409809442718879744).

Other three tests are failing since July 9th 2021. Commit 617064d73, infra commit 27c63cf72, [build](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/1413428671103897600).

#### Testgrid link:
Testgrid: https://k8s-testgrid.appspot.com/sig-release-master-informing#gce-cos-master-serial&width=20
Prow history: https://prow.k8s.io/job-history/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial

#### Reason for failure:
For CSI Volumes:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/subpath.go:348
Jul  9 10:01:28.012: while cleaning up resource
Unexpected error:
    <errors.aggregate | len:1, cap:1>: [
        [
            {
                msg: "persistent Volume pvc-8c22a1cc-d586-4f3b-a6ce-eba8c7425fa6 not deleted by dynamic provisioner: PersistentVolume pvc-8c22a1cc-d586-4f3b-a6ce-eba8c7425fa6 still exists within 5m0s",
                err: {
                    s: "PersistentVolume pvc-8c22a1cc-d586-4f3b-a6ce-eba8c7425fa6 still exists within 5m0s",
                },
            },
        ],
    ]
    persistent Volume pvc-8c22a1cc-d586-4f3b-a6ce-eba8c7425fa6 not deleted by dynamic provisioner: PersistentVolume pvc-8c22a1cc-d586-4f3b-a6ce-eba8c7425fa6 still exists within 5m0s
occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/subpath.go:183
```

For In-tree Volumes:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/subpath.go:341
Jul  9 14:04:08.625: Expected grep stdout to be empty (i.e. no subpath mount found).
Expected
    <string>: 963 355 0:40 /provisioning-7593/provisioning-7593 /var/lib/kubelet/pods/1cab2854-f6e2-400c-9136-9986d769b168/volume-subpaths/test-volume/test-container-subpath-inlinevolume-ttr8/0 rw shared:27 - tmpfs tmpfs rw
    966 125 0:40 /provisioning-7593/provisioning-7593 /var/lib/kubelet/pods/1cab2854-f6e2-400c-9136-9986d769b168/volume-subpaths/test-volume/test-container-subpath-inlinevolume-ttr8/0 rw,nosuid,nodev,noexec shared:27 - tmpfs tmpfs rw
    965 383 0:40 /provisioning-7593/provisioning-7593 /home/kubernetes/containerized_mounter/rootfs/var/lib/kubelet/pods/1cab2854-f6e2-400c-9136-9986d769b168/volume-subpaths/test-volume/test-container-subpath-inlinevolume-ttr8/0 rw,nosuid,nodev,noexec shared:27 - tmpfs tmpfs rw
    964 381 0:40 /provisioning-7593/provisioning-7593 /home/kubernetes/containerized_mounter/rootfs/var/lib/kubelet/pods/1cab2854-f6e2-400c-9136-9986d769b168/volume-subpaths/test-volume/test-container-subpath-inlinevolume-ttr8/0 rw,nosuid,nodev,noexec shared:27 - tmpfs tmpfs rw
    
to be empty
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/utils/utils.go:202
```

#### Anything else we need to know:

There is another open issues for failing tests in this job: https://github.com/kubernetes/kubernetes/issues/102865 Failures reported there are about missing metrics, and were occuring before unmounting problems. So I opened a new issue, even though it's the same SIG. These tests don't look related to me, but I don't have enough context, maybe they somehow are.

/sig storage
/kind failing-test
/milestone 1.22

### PR DESCRIPTION
**Pull request by jingxu97**
1. fix command empty issue for some Windows storage tests
2. enable more windows storage tests by adding ntfs test patten

Change-Id: Ic33be282d669a23107474a14d4368bbf95c9b459

#### Which issue(s) this PR fixes:

Fixes #103651

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/pod/create.go [hunk 1]
```diff
@@ -166,8 +166,12 @@ func MakeSecPod(podConfig *Config) (*v1.Pod, error) {
 	if podConfig.NS == "" {
 		return nil, fmt.Errorf("Cannot create pod with empty namespace")
 	}
-	if len(podConfig.Command) == 0 && !NodeOSDistroIs("windows") {
-		podConfig.Command = "trap exit TERM; while true; do sleep 1; done"
+	if len(podConfig.Command) == 0 {
+		if !NodeOSDistroIs("windows") {
+			podConfig.Command = "trap exit TERM; while true; do sleep 1; done"
+		} else {
+			podConfig.Command = "while true; do sleep 1; done"
```
Comment (neolit123) [inline test/e2e/framework/pod/create.go:173]: what shell is running on the Windows nodes?
given `while true; do sleep 1; done` i would assume its a Unix-like shell, but why `trap exit TERM; ` can't be used?

Comment (jingxu97) [reply test/e2e/framework/pod/create.go:173]: I am not very sure, from previous testing, it does not work well for Windows.

Comment (jingxu97) [reply test/e2e/framework/pod/create.go:173]: I will try again, and update the details

Comment (jingxu97) [reply test/e2e/framework/pod/create.go]: actually I tried "trap exist TERM" on Windows, it is working fine now. Updated the code

**Review by neolit123 — COMMENTED** [review]
the CI is failing with storage tests, possibly related?
/approve
(for the framework change, but ideally sig-windows/storage should have another look)

**Comment by puerco** [comment]
@jingxu97 are you planning on cherry-picking this change to `release-1.22` ?

**Comment by saschagrunert** [comment]
It does not look like that the change affects anything on release-1.22 [blocking](https://testgrid.k8s.io/sig-release-1.22-blocking) or [informing](https://testgrid.k8s.io/sig-release-1.22-informing) dashboards. So we're probably good to not cherry-pick this one.

**Pull request merged**